app/views/qing_groups/_modal.html.erb

```diff
@@ -4,10 +4,12 @@
       <div class="modal-header">
         <button type="button" class="close group-cancel" data-dismiss="modal" aria-hidden="true">&times;</button>
         <h4 class="modal-title">
-          <% if @qing_group.group_name %>
+          <% if @qing_group.group_name && action_name == "edit" %>
             <%= I18n.t("activerecord.attributes.qing_group.edit") %>
           <% elsif @qing_group.new_record? %>
             <%= I18n.t("activerecord.attributes.qing_group.add") %>
+          <% elsif action_name == "show" %>
+            <%= I18n.t("activerecord.attributes.qing_group.show", name: @qing_group.group_name) %>
```

Comment:
Decorator! Could do this in one call to `I18n.t` with `action_name` interpolated, like `I18n.t("activerecord.attributes.qing_group.#{action_name}", name: @qing_group.group_name)`. You can always pass the `name` param and it will just be ignored in some cases.

Be sure to manual test all three after the refactor!

Reply:
You may need to rename the 'add' i18n key to 'new' to get it to work with action_name